pkg/machinepool/machinepool.go

```diff
@@ -29,6 +33,49 @@ func NewMachinePoolService() MachinePoolService {
 	return &machinePool{}
 }
 
+// ListMachinePools lists all machinepools (or, nodepools if hypershift) in a cluster
+func (m *machinePool) ListMachinePools(r *rosa.Runtime, clusterKey string, cluster *cmv1.Cluster, isHypershift bool) {
+	// Load any existing machine pools for this cluster
+	r.Reporter.Debugf("Loading machine pools for cluster '%s'", clusterKey)
+	var err error
+	var machinePools []*cmv1.MachinePool
+	var nodePools []*cmv1.NodePool
+	if !isHypershift {
+		machinePools, err = r.OCMClient.GetMachinePools(cluster.ID())
+	} else {
+		nodePools, err = r.OCMClient.GetNodePools(cluster.ID())
+	}
+	if err != nil {
+		r.Reporter.Errorf("Failed to get machine pools for cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	if output.HasFlag() {
+		if !isHypershift {
+			err = output.Print(machinePools)
+		} else {
+			err = output.Print(nodePools)
+		}
+		if err != nil {
+			r.Reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+		os.Exit(0)
```

Comment:
Based on the comment above, best we return an error or nil 

```suggestion
		if err != nil {
			return err
		}
		return nil
```